### PR TITLE
Make ticklabel_format work both for 2D and 3D axes.

### PR DESCRIPTION
**Pull request by anntzer**
... instead of duplicating its implementation in axes3d with just minor
changes.

Note that ticklabel_format was not used anywhere either in the tests or
in the examples.  Exercise it in the scalarformatter example, which also
gets slightly updated -- ScalarFormatter has been the default formatter
for years, so doesn't really warrant the label "The new formatter".

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/axes/_base.py [hunk 1]
```diff
@@ -2866,35 +2884,24 @@ def ticklabel_format(self, *, axis='both', style='', scilimits=None,
             sb = None
         else:
             raise ValueError("%s is not a valid style value")
+        axis_map = {**self._get_axis_map(), 'both': self._get_axis_list()}
+        cbook._check_in_list(axis_map, axis=axis)
```
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: `_check_getitem` now?

File: lib/matplotlib/axes/_base.py [hunk 2]
```diff
@@ -2866,35 +2884,24 @@ def ticklabel_format(self, *, axis='both', style='', scilimits=None,
             sb = None
         else:
             raise ValueError("%s is not a valid style value")
+        axis_map = {**self._get_axis_map(), 'both': self._get_axis_list()}
```
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py]: `_get_axis_map` doesn't return a `Str: List[Axis]`, does it? It looks like it's `Str: Axis`.

Comment (anntzer) [reply lib/matplotlib/axes/_base.py]: good catch, thanks, fixed.

**Comment by jklymak** [comment]
Needs tests?

**Comment by anntzer** [comment]
test's in

**Comment by QuLogic** [comment]
No 3D tests? This is labelled `mplot3d` after all...

**Comment by anntzer** [comment]
fair enough, moved the test to test_mplot3d

**Pull request merged**